daffodil-core/src/main/scala/org/apache/daffodil/dsom/IIBase.scala

```diff
@@ -167,11 +169,11 @@ abstract class IIBase( final override val xml: Node, xsdArg: XMLSchemaDocument,
    * we will detect it.
    */
   protected final lazy val seenBeforeThisFile: IIMap = LV('seenBeforeThisFile) {
-    val res = Delay {
+    val res = Delay( {
       val v = if (notSeenThisBefore) seenBefore.value + mapTuple
       else seenBefore.value
       v
-    }
+    }, 'seenBeforeThisFile)
```

Comment:
Just a minor stylistic thing, thoughts on swapping the order of the Delay parameters so this would look like this instead:
```scala
    val res = Delay('seenBeforeThisFile, {
      val v = if (notSeenThisBefore) seenBefore.value + mapTuple
      else seenBefore.value
      v
    })
```
That feels a bit easier to read to me.